packages/toast/src/Toast.tsx

```diff
@@ -0,0 +1,335 @@
+import React, { useRef } from 'react';
+import { Transition } from 'react-transition-group';
+import { transparentize } from 'polished';
+import { cx, css, keyframes } from '@leafygreen-ui/emotion';
+import Portal from '@leafygreen-ui/portal';
+import { uiColors } from '@leafygreen-ui/palette';
+import { spacing } from '@leafygreen-ui/tokens';
+import { Body } from '@leafygreen-ui/typography';
+import IconButton from '@leafygreen-ui/icon-button';
+import CheckmarkWithCircleIcon from '@leafygreen-ui/icon/dist/CheckmarkWithCircle';
+import RefreshIcon from '@leafygreen-ui/icon/dist/Refresh';
+import XIcon from '@leafygreen-ui/icon/dist/X';
+
+const progressBackgroundBase = '#22B7EB';
+const toastWidth = 400;
+
+const progressBarBackgroundStyle = css`
+  position: absolute;
+  bottom: 0;
+  left: 0;
+  right: 0;
+  height: 6px;
+  background-color: ${uiColors.gray.light2};
+  border-radius: 0 0 4px 4px;
+  overflow: hidden;
+`;
+
+const backgroundShimmer = keyframes`
+  0% {
+    background-position: ${-toastWidth}px;
+  }
+
+  100% {
+    background-position: ${toastWidth * 2}px;
+  }
+`;
+
+const progressBarStyle = css`
+  overflow: hidden;
+  height: 6px;
+  position: absolute;
+  top: 0;
+  bottom: 0;
+  left: 0;
+  background-color: ${progressBackgroundBase};
+  background-image: linear-gradient(
+    90deg,
+    ${progressBackgroundBase} 0px,
+    #cce8f4 ${toastWidth / 2}px,
+    ${progressBackgroundBase} ${toastWidth}px
+  );
+  background-size: 600px;
+  animation: ${backgroundShimmer} 4s infinite linear;
+  transition: width 0.3s ease-in-out;
+`;
+
+interface ProgressBarProps {
+  progress: number;
+}
+
+function ProgressBar({ progress }: ProgressBarProps) {
```

Comment:
do you think this function and the surrounding logic could be it's own package? we can leave it here for now if you prefer, but I can definitely see us wanting to a) standardize the usage of progress bars and b) share this across components as we work on larger loaders

Reply:
This probably won't be its own package. The implementation is super specific to this component. That said, I think I'll break it into it's own file in the package...